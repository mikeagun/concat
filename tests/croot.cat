#Copyright (C) 2024 D. Michael Agun
#
#Licensed under the Apache License, Version 2.0 (the "License");
#you may not use this file except in compliance with the License.
#You may obtain a copy of the License at
#
#http://www.apache.org/licenses/LICENSE-2.0
#
#Unless required by applicable law or agreed to in writing, software
#distributed under the License is distributed on an "AS IS" BASIS,
#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#See the License for the specific language governing permissions and
#limitations under the License.

[1.0 3/^] \croot def
# [      - start of code block (so now we aren't executing code but collecting code into a block for later execution)
# 1.0    - push 1 as a floating point number (otherwise the / that comes later would do integer divide)
# 3 / ^  - push 3, then divide 1 by 3 to get 1/3, then take cubed root of top of stack (by raising to 1/3)
# ]      - end of code block (so at this point the code block [1.0 3/^] is currently the top of the stack)
# \croot - escaped word (so instead of trying to execute croot as an operator, it will push the operator name onto the stack)
# def - define a new operator with top as the word and second as the definition

#now we can use croot like any other operator
4 croot print
8 croot print
100 croot print
1000 croot print
