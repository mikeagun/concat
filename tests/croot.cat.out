1.587401
2.000000
4.641589
10.000000
